File: java/statePattern/MainEngine.java
package statePattern;

public class MainEngine implements State {
	private MissionControl missionControl;

	public MainEngine(MissionControl missionControl) {
		this.missionControl = missionControl;
	}

	@Override
	public void checkAllSystems() {
		System.out.println("Systems have been cleared");
	}

	@Override
	public void startPowerUnits() {
		System.out.println("Power units have been started");
		this.missionControl.setState(this.missionControl.getRetractArms());
	}

	@Override
	public void retractArms() {
		System.out.println("Arms and fuel connections have been retracted.");

	}

	@Override
	public void shutDown() {
		System.out.println("All systems are to cease operations immediatly");
		this.missionControl.setState(this.missionControl.getAbortProcedure());
	}

	@Override
	public void rocketBooster() {
		System.out.println("Power units have not been started.");

	}

}

File: php/statePattern/MainEngine.php
<?php  namespace php\statePattern;

class MainEngine implements State {
	
	/**
	 * Mission control instance
	 *
	 * @var MissionControl $missionControl
	 */
	public $missionControl;
		
	/**
	 * Set instance for mission control
	 *
	 * @param MissionControl $missionControl
	 * @return void
	 */
	public function __construct(MissionControl $missionControl) {
		$this->missionControl = $missionControl;
	}
	
	/**
	 * Check all systems are stable and ready to go
	 *
	 * @return void
	 */
	public function checkAllSystems() {
		echo "Systems have already been cleared.<br>";
	}
	
	/**
	 * set state in mission control
	 *
	 * @return void
	 */
	public function startPowerUnits() {
		echo "power units started.<br>";
		$this->missionControl->setState($this->missionControl->getRetractArms());
	}
	
	public function retractArms() {
		echo "Power units have not been started.<br>";
	}
	
	public function shutDown() {
		echo "All systems are to cease operations immediatly.<br>";
		$this->missionControl->liftOffHandOver($this->missionControl->getAbortProcedure());
	}
	
	public function rocketBooster() {
		echo "power units have not been started.<br>";
	}
}
